### ISSUE TITLE
Detect language

### ISSUE THREAD

**Issue by Bunn** [Bunn/macGist#3]
Detect programming language to add correct gist extension.

**Comment by fpg1503** [comment]
Github's [linguist](https://github.com/github/linguist) is open source, you could try calling it using CRuby.

**Comment by Bunn** [comment]
@fpg1503 yep, I saw that, it's on my bookmark already, there's also some cool implementations using ML like: https://medium.com/machine-learning-at-petiteprogrammer/detecting-the-programming-language-of-source-code-snippets-using-machine-learning-and-neural-e275ca4ce93e

Not sure how to handle this yet, but probably the linguist approach is the best one.
Thanks :)
